### PR TITLE
[MIG] stock_by_warehouse: Migration to 17.0 T#79454

### PR DESCRIPTION
**Pull request by xmglord**
- Simplify the way to define modifiers as states, required, readonly,
  invisible and column_invisible as part of [1].
- Remove owl="1" from OWL templates, as it's not needed anymore as all
  of them are OWL now as part of [2].
- Change qty_done to quantity and picked = True as part of [3].

[1] odoo/odoo#104741
[2] odoo/odoo#130467
[3] odoo/odoo#137864

### CODE REVIEW

**Comment by xmglord** [comment]
@luisg123v please review.

**Comment by luisg123v** [comment]
Missing task ID on PR's title.

File: stock_by_warehouse/views/product_product_views.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            <xpath expr="//field[@name='default_code']" position="before">
+                <field
+                    name="warehouses_stock"
+                    widget="warehouse"
+                    groups="stock.group_stock_multi_warehouses"
+                    invisible="detailed_type != 'product'"/>
+                <label
+                    for="warehouses_stock_location"
+                    groups="stock.group_stock_multi_warehouses"
+                    invisible="detailed_type != 'product'"/>
```
Comment (luisg123v) [inline stock_by_warehouse/views/product_product_views.xml]: Indentation is not correct here, missing running precommit.

File: stock_by_warehouse/static/src/components/warehouse/warehouse_field.xml [hunk 2]
```diff
@@ -0,0 +1,116 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<templates xml:space="preserve">
+
+    <t t-name="stock_by_warehouse.ShowWarehouseInfo">
```
Comment (luisg123v) [inline stock_by_warehouse/static/src/components/warehouse/warehouse_field.xml:4]: Why this change? Please document in commit description.

Some examples for the documentation part:
- https://github.com/OCA/hr/commit/4fb14e7de120b46c3bf06bf6a4e6b533e0c82d28
- https://github.com/OCA/account-financial-tools/commit/ec5ec0ee0d5534f502d010e0e3329bcde1c87b71

**Review by luisg123v — CHANGES_REQUESTED** [review]
CI is red.

File: stock_by_warehouse/views/product_product_views.xml (hunk 1 above)
Comment (xmglord) [reply stock_by_warehouse/views/product_product_views.xml]: done

File: stock_by_warehouse/static/src/components/warehouse/warehouse_field.xml (hunk 2 above)
Comment (xmglord) [reply stock_by_warehouse/static/src/components/warehouse/warehouse_field.xml:4]: done

**Comment by xmglord** [comment]
@luisg123v please review, comments applied and CI is green.

**Comment by luisg123v** [comment]
FYI @xmglord you can click the "Re-request review" button.

File: stock_by_warehouse/models/product_product.py [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+        return domain_quant
+
+    @api.depends_context("warehouse", "company")
+    def _compute_qty_available_not_reserved(self):
+        """Method to set the quantity available that is not reserved of a serie of products."""
+        domain_quant = self._prepare_domain_available_not_reserved()
+        quants = (
+            self.env["stock.quant"]
+            .with_context(lang=False)
+            .read_group(domain_quant, fields=["product_id", "quantity", "reserved_quantity"], groupby="product_id")
```
Comment (luisg123v) [inline stock_by_warehouse/models/product_product.py]: Please read [my comment here](https://github.com/OCA/odoo-module-migrator/issues/90#issuecomment-2008472875)

**Review by luisg123v — CHANGES_REQUESTED** [review]

File: stock_by_warehouse/models/product_product.py [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+                domain_quant,
+                groupby=["product_id"],
+                aggregates=["quantity:sum", "reserved_quantity:sum"],
+            )
+        )
+
+        res = {}
+        for quant in quants:
+            quantity = quant[1] - quant[2]
+            res[quant[0].id] = max(quantity, 0.0)
```
Comment (luisg123v) [inline stock_by_warehouse/models/product_product.py]: These are harder to read.

Maybe:

`for product, quantity, reserved_quantity in quants:`

In fact, now using dict comprehension could be clearer

Comment (xmglord) [reply stock_by_warehouse/models/product_product.py]: done

Comment (luisg123v) [reply stock_by_warehouse/models/product_product.py]: By dict comprehension, I meant:

```
res = {
    product.id: max(quantity - reserved_quantity, 0.0)
    for product, quantity, reserved_quantity in quants
}
```

That would be faster because creating a filled dictionary is faster than creating an empty one and updating one key a time.

Comment (xmglord) [reply stock_by_warehouse/models/product_product.py]: done, sorry i miss understood.

**Review by luisg123v — APPROVED** [review]
LGTM :+1:
